addons/point_of_sale/models/pos_order.py

```diff
@@ -358,8 +357,22 @@ def action_view_invoice(self):
 
     def action_pos_order_paid(self):
         self.ensure_one()
-        if not float_is_zero(self.amount_total - self.amount_paid, precision_rounding=self.currency_id.rounding):
+
+        if not self.config_id.cash_rounding and not float_is_zero(self.amount_total - self.amount_paid, precision_rounding=self.currency_id.rounding):
             raise UserError(_("Order %s is not fully paid.") % self.name)
+        elif self.config_id.cash_rounding:
+            rounding = self.config_id.rounding_method.rounding
```

Comment:
You should call float_round, you can specify the rounding method ('UP', 'DOWN', 'HALF-UP'), and it will make only one call

Reply:
just round the total with correct method and compare  it to amount_paid